### ISSUE TITLE
size-adjust

### ISSUE THREAD

**Issue by firefoxic** [web-platform-tests/interop#542]
### Description

The `size-adjust` descriptor for the `@font-face` at-rule defines a multiplier for glyph outlines and metrics associated with this font, to allow the author to harmonize the designs of various fonts when rendered at the same `font-size`.

### Specification

https://drafts.csswg.org/css-fonts-5/#size-adjust-desc

### Tests

https://wpt.fyi/results/css/css-fonts/size-adjust-01.html
https://wpt.fyi/results/css/css-fonts/size-adjust-02.html
https://wpt.fyi/results/css/css-fonts/size-adjust-03.html
https://wpt.fyi/results/css/css-fonts/size-adjust-text-decoration.tentative.html
https://wpt.fyi/results/css/css-fonts/size-adjust-unicode-range-system-fallback.html
https://wpt.fyi/results/css/css-fonts/size-adjust.tentative.html

### Current Implementations

- [X] Blink
- [X] Gecko
- [x] WebKit

### Browser bug reports

https://bugs.webkit.org/show_bug.cgi?id=256081

### Other

https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/size-adjust
https://caniuse.com/mdn-css_at-rules_font-face_size-adjust

**Comment by shivamidow** [comment]
I think WebKit already implemented this feature and shipped it in Safari 17 although not all tests are green. Check your setting `Develop > Feature Flags` after enabling the developer option.

**Comment by nt1m** [comment]
This is already implemented and mostly interoperable across browsers. Safari implemented in v17 and Safari Technology Preview.

There are 2 minor bugs that differ:
* Firefox & Safari fail: [size-adjust-unicode-range-system-fallback.html](https://wpt.fyi/results/css/css-fonts/size-adjust-unicode-range-system-fallback.html?label=experimental&label=master&aligned)
* Safari fails: [size-adjust-text-decoration.tentative.html](https://wpt.fyi/results/css/css-fonts/size-adjust-text-decoration.tentative.html?label=experimental&label=master&aligned), although this is pretty trivial to fix and doesn't necessarily prevent using the feature.

**Comment by foolip** [comment]
https://wpt.fyi/results/css/css-fonts/size-adjust-01.html is failing in Chrome, and that failure is linked to https://crbug.com/451346. But that's about `font-size-adjust`, not sure if the failure is linked to the wrong bug or what.

**Comment by nt1m** [comment]
The Safari failure on https://wpt.fyi/results/css/css-fonts/size-adjust-text-decoration.tentative.html is due to https://bugs.webkit.org/show_bug.cgi?id=262866 which is orthogonal to the size-adjust functionality.

**Comment by nt1m** [comment]
> https://wpt.fyi/results/css/css-fonts/size-adjust-01.html is failing in Chrome, and that failure is linked to https://crbug.com/451346. But that's about `font-size-adjust`, not sure if the failure is linked to the wrong bug or what.

It looks like the Chrome failure is just fuzziness.

**Comment by dholbert** [comment]
> * Firefox & Safari fail: [size-adjust-unicode-range-system-fallback.html](https://wpt.fyi/results/css/css-fonts/size-adjust-unicode-range-system-fallback.html?label=experimental&label=master&aligned)

At least for Firefox, we're only failing on the wpt.fyi CI for some reason. The test passes in our own CI, and it passes locally on my machine. I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1860124 to investigate.

**Comment by dholbert** [comment]
The other Firefox failure here is [size-adjust.tentative.html](https://wpt.fyi/results/css/css-fonts/size-adjust.tentative.html), tracked in [bug 1837533](https://bugzilla.mozilla.org/show_bug.cgi?id=1837533), thought the failure there seems to be a subtle vertical-spacing thing -- to-be-determined whether it's a Firefox bug or a test bug, but in any case, that test-failure is unrelated to the `size-adjust` feature.

**Comment by gsnedders** [comment]
See https://wpt.fyi/results/css/css-fonts?label=master&label=experimental&aligned&q=size-adjust%20%21font-size-adjust for all the tests.

**Comment by shivamidow** [comment]
> https://wpt.fyi/results/css/css-fonts/size-adjust-01.html is failing in Chrome, and that failure is linked to https://crbug.com/451346. But that's about `font-size-adjust`, not sure if the failure is linked to the wrong bug or what.

I think the failure indicates a wrong pointer. `font-size-adjust` is irrelevant to that test. Also, the failure on size-adjust-01.html seems limited to the Chrome product. When I tested it with a Chromium build on Linux, the test passed.

I found that size-adjust-01.html.ini has the following lines. Perhaps, the Google Chrome team has a better understanding about this.
```
[size-adjust-01.html]                                                                                                                                                     
  expected:                                                                                                                                                               
    if product == "chrome": FAIL
```

**Comment by foolip** [comment]
@shivamidow the size-adjust-01.html.ini file is just a way of recording that the failure is known and these files are generated automatically. We'd need to take a closer look at size-adjust-01.html, but as @nt1m points out it's just a very minor difference between expected and actual, so probably not very impactful.

**Comment by jgraham** [comment]
Thank you for proposing size-adjust for inclusion in Interop 2024.

We wanted to let you know that this proposal was not selected to be part of Interop this year.

This proposal was not included in Interop 2024 because when we looked at its Web Platform Tests results, we found that it is already largely interoperable between tested browsers and does not need special coordinated attention.

For an overview of our process, see [proposal selection](https://github.com/web-platform-tests/interop/blob/main/2024/selection-process.md). Thank you again for contributing to Interop 2024!

_Posted on behalf of the Interop team._
